tools/ctl/ide/ui.go

```diff
@@ -216,6 +227,78 @@ func (e *explorer) showLogs(ctx context.Context, example firestore.Rebuild) {
 	}
 }
 
+func (e *explorer) editAndRun(ctx context.Context, example firestore.Rebuild) {
+	localAssets, err := localAssetStore(ctx, example.Run)
+	if err != nil {
+		log.Println(errors.Wrap(err, "failed to create local asset store"))
+		return
+	}
+	buildDefAsset := rebuild.Asset{Type: rebuild.BuildDef, Target: example.Target()}
+	var currentStratYaml []byte
```

Comment:
I'd change this type to be a strategyoneof. you can encode it to the asset writer below. instead of marshalling it here.